test/ILLink.RoslynAnalyzer.Tests/TestChecker.cs

```diff
@@ -271,43 +272,68 @@ bool Matches (Diagnostic diagnostic)
 
 		private bool TryValidateLogContainsAttribute (AttributeSyntax attribute, List<Diagnostic> diagnostics, out int? matchIndex, out string? missingDiagnosticMessage)
 		{
-			missingDiagnosticMessage = null;
-			matchIndex = null;
+			if (!LogContains(attribute, diagnostics, out matchIndex, out string text)) {
+				missingDiagnosticMessage = $"Could not find text:\n{text}\nIn diagnostics:\n{string.Join (Environment.NewLine, _diagnostics)}";
+				return false;
+			}
+			else {
+				missingDiagnosticMessage = null;
```

Comment:
Unnecessary - it's already assigned null

Reply:
I removed the initialization above, the build fails without this assignment.